proposals/0020/README.md

```diff
@@ -0,0 +1,157 @@
+        timeout: 600
+        environment:
+          MIRROR_HOST: <MIRROR_HOST_IP>
+        volumes:
+          - /statedir:/statedir
+      - name: install-root-fs
+        image: install-root-fs:v1.0.42
+        timeout: 600
+...
+```  
+
+Structure in which the above template will be parsed should look like this:
+
+```go
+type Workflow struct {
+        Version       string `yaml:"version"`
+        Name          string `yaml:"name"`
+        ID            string `yaml:"id"`
+        GlobalTimeout int    `yaml:"global_timeout"`
+        Tasks         []Task `yaml:"tasks"`
+}
+
+```
+
+### 2. Template after proposal [0013]
+
+When proposal [0013] is implemented a new field in the template called as `service` will be introduced and the template will look like as following:
+
+```yaml
+version: "0.2"
+name: ubuntu_provisioning
+global_timeout: 6000
+services:
+  - name: serve boot files
+    worker: {{provisioner1}}
+    image: osie-boot-files-server:v1.0.42
+    labels:
+      - "traefik.http.routers.myrouter.rule=Host(`{{workflowid}}.workflows.tinkerbell.local`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/kernel`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/initrd`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/modules`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/auto.ipxe`)"
+
+  - name: password receiver
+    worker: "{{provisioner1}}"
+    image: password-receiver:v1.0.42
+    labels:
+      - "traefik.http.routers.myrouter.rule=Host(`{{workflowid}}.workflows.tinkerbell.local`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/key`)"
+      - "traefik.http.routers.myrouter.rule=Path(`/password`)"
+
+tasks:
+  - name: os-installation
+    worker: {{.device_1}}
+    volumes:
+      - /dev:/dev
+      - /dev/console:/dev/console
+      - /lib/firmware:/lib/firmware:ro
+    environment:
+      MIRROR_HOST: <MIRROR_HOST_IP>
+    actions:
+      - name: disk-wipe
+        image: disk-wipe:v1.0.42
+        timeout: 90
+      - name: disk-partition
+        image: disk-partition:v1.0.42
+        timeout: 600
+        environment:
+          MIRROR_HOST: <MIRROR_HOST_IP>
+        volumes:
+          - /statedir:/statedir
+      - name: install-root-fs
+        image: install-root-fs:v1.0.42
+        timeout: 600
+...
+```
+
+Since there is a new field `services` inserted in the above template, There will be a new field needs to be added in the structre to parse the above template which will be as follows:
+
+```go
+type Workflow struct {
+        Version       string `yaml:"version"`
+        Name          string `yaml:"name"`
+        ID            string `yaml:"id"`
+        GlobalTimeout int    `yaml:"global_timeout"`
+        Services      []Service `yaml:"services"`
+        Tasks         []Task `yaml:"tasks"`
+}
+
+```
+
+Now if we would like to support both of the above templates then we need to identify the struct in which we need to parse the template which we can identify based on the versions.
```

Comment:
Kubernetes handles this by having a storage version (that is rich enough to store data about all supported versions) and allowing the user to specify which version they are providing, we can likely do similar here, and since from what I can tell the changes appear purely additive in nature, we could also use the new version as the storage version.